library/src/androidTest/java/org/xmtp/android/library/DmTest.kt

```diff
@@ -344,16 +348,69 @@ class DmTest {
         runBlocking {
             val dm =
                 boClient.conversations.findOrCreateDm(alix.walletAddress)
-            assert(boClient.contacts.isGroupAllowed(dm.id))
+            assert(boClient.contacts.isConversationAllowed(dm.id))
             assertEquals(dm.consentState(), ConsentState.ALLOWED)
 
-            boClient.contacts.denyGroups(listOf(dm.id))
-            assert(boClient.contacts.isGroupDenied(dm.id))
+            boClient.contacts.denyConversations(listOf(dm.id))
+            assert(boClient.contacts.isConversationDenied(dm.id))
             assertEquals(dm.consentState(), ConsentState.DENIED)
 
             dm.updateConsentState(ConsentState.ALLOWED)
-            assert(boClient.contacts.isGroupAllowed(dm.id))
+            assert(boClient.contacts.isConversationAllowed(dm.id))
+            assertEquals(dm.consentState(), ConsentState.ALLOWED)
+        }
+    }
+
+    @Test
+    fun testSyncConsent() {
+        val key = SecureRandom().generateSeed(32)
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        val alixWallet = PrivateKeyBuilder()
+
+        val alixClient = runBlocking {
+            Client().createV3(
+                account = alixWallet,
+                options = ClientOptions(
+                    ClientOptions.Api(XMTPEnvironment.LOCAL, false),
+                    enableV3 = true,
+                    appContext = context,
+                    dbEncryptionKey = key
+                )
+            )
+        }
+        runBlocking {
+            val dm =
+                alixClient.conversations.findOrCreateDm(bo.walletAddress)
+            assert(alixClient.contacts.isConversationAllowed(dm.id))
             assertEquals(dm.consentState(), ConsentState.ALLOWED)
         }
+        alixClient.dropLocalDatabaseConnection()
+        alixClient.deleteLocalDatabase()
+
+        val alixClient2 = runBlocking {
+            Client().createV3(
+                account = alixWallet,
+                options = ClientOptions(
+                    ClientOptions.Api(XMTPEnvironment.LOCAL, false),
+                    enableV3 = true,
+                    appContext = context,
+                    dbEncryptionKey = key
+                )
+            )
+        }
+
+        val state = runBlocking { alixClient2.inboxState(true) }
+        assertEquals(state.installations.size, 2)
+
+        runBlocking {
+            alixClient2.conversations.syncConversations()
+            val dm2 =
+                alixClient2.conversations.findOrCreateDm(bo.walletAddress)
+            alixClient2.syncConsent()
```

Comment:
@codabrink I'm getting `uniffi.xmtpv3.GenericException$DeviceSync: group error: group not found` whenever calling the consent sync method. Am I doing this wrong? It should be using the local hosted history server `http://0.0.0.0:5558` https://github.com/xmtp/xmtp-message-history-server I'm running that locally.

Reply:
Ah, you need to enable history sync first to create the group. It's something I kept from the old flow, but I don't like the design of it. Let me adjust this so it'll auto-create the group in the upcoming PR so you don't run into this.